stdlib/src/builtin/bin.mojo

```diff
@@ -0,0 +1,81 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+"""Implements the `bin()` function
+
+These are Mojo built-ins, so you don't need to import them.
+"""
+
+
+@always_inline("nodebug")
+fn bin(b: Scalar[DType.bool]) -> String:
+    """Returns the binary representation of a scalar bool.
+
+    Args:
+        b: A scalar bool value.
+
+    Returns:
+        The binary string representation of b.
+    """
+    return bin(int(b))
+
+
+fn bin[type: DType](num: Scalar[type]) -> String:
```

Comment:
Python declares this with a positional-only argument, and I think that's a good idea. Could you please change these (and you can do the same to the `index` function).
```suggestion
fn bin[type: DType](num: Scalar[type], /) -> String:
```

Reply:
Oh TIL, yeah I definitely do that